Bug: Regression: Running Mimir 2.16.0 with target: all,query-scheduler and service_discovery_mode: ring results in a panic

### What is the bug?

ts=2025-05-11T10:08:25.032667977Z caller=featurecontrol.go:97 level=warn msg="Classic mode enabled"
panic: duplicate metrics collector registration attempted

goroutine 1 [running]:
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0x4ff13a0, {0xc000afda30?, 0x0?, 0x0?})
	/__w/mimir/mimir/vendor/github.com/prometheus/client_golang/prometheus/registry.go:406 +0x66
github.com/prometheus/client_golang/prometheus/promauto.Factory.NewCounterVec({{0x36fcf30?, 0x4ff13a0?}}, {{0x0, 0x0}, {0x0, 0x0}, {0x3061164, 0x3d}, {0x3056891, 0x39}, ...}, ...)
	/__w/mimir/mimir/vendor/github.com/prometheus/client_golang/prometheus/promauto/auto.go:276 +0x163
github.com/grafana/mimir/pkg/util.NewRequestInvalidClusterValidationLabelsTotalCounter({0x36fcf30, 0x4ff13a0}, {0x2f9a0cf, 0xe}, {0x2f855e2, 0x4})
	/__w/mimir/mimir/pkg/util/metrics.go:23 +0x1bc
github.com/grafana/mimir/pkg/scheduler.NewScheduler({0x64, 0x0, {0x6400000, 0x6400000, {0x0, 0x0}, 0x0, 0x0, 0x0, {0x5f5e100, ...}, ...}, ...}, ...)
	/__w/mimir/mimir/pkg/scheduler/scheduler.go:163 +0x689
github.com/grafana/mimir/pkg/mimir.(*Mimir).initQueryScheduler(0xc000a1c000)
	/__w/mimir/mimir/pkg/mimir/modules.go:1095 +0xa8
github.com/grafana/dskit/modules.(*Manager).initModule(0xc000639230, {0xc00081aef1, 0xf}, 0xc000b9b758, 0xc000a5f530)
	/__w/mimir/mimir/vendor/github.com/grafana/dskit/modules/modules.go:136 +0x1ea
github.com/grafana/dskit/modules.(*Manager).InitModuleServices(0xc000639230, {0xc000a5fe00, 0x3, 0xf?})
	/__w/mimir/mimir/vendor/github.com/grafana/dskit/modules/modules.go:108 +0xe8
github.com/grafana/mimir/pkg/mimir.(*Mimir).Run(0xc000a1c000)
	/__w/mimir/mimir/pkg/mimir/mimir.go:902 +0x387
main.main()
	/__w/mimir/mimir/cmd/mimir/main.go:227 +0xf85

### How to reproduce it?

Start mimir 2.16 with a configuration of target: all,query-scheduler
mimir -target=all,query-scheduler -query-scheduler.service-discovery-mode=ring

### What did you think would happen?

To correctly start mimir

### What was your environment?

Kubernetes, mimir statefulset in monolithic mode

### Any additional context to share?

This works correctly in versions prior to 2.16.0

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -57,6 +57,7 @@
 * [BUGFIX] Query-frontend: Fix an issue where transient errors could be inadvertently cached. #11198
 * [BUGFIX] Ingester: read reactive limiters should activate and deactivate when the ingester changes state. #11234
 * [BUGFIX] Query-frontend: Fix an issue where errors from date/time parsing methods did not include the name of the invalid parameter. #11304
+* [BUGFIX] Query-frontend: Fix a panic in monolithic mode caused by a clash in labels of the `cortex_client_invalid_cluster_validation_label_requests_total` metric definition. #11455
 * [BUGFIX] Compactor: Fix issue where `MimirBucketIndexNotUpdated` can fire even though the index has been updated within the alert threshold. #11303
 
 ### Mixin

--- a/pkg/blockbuilder/metrics.go
+++ b/pkg/blockbuilder/metrics.go
@@ -60,7 +60,7 @@ func newBlockBuilderMetrics(reg prometheus.Registerer) blockBuilderMetrics {
 		Help: "Total number of blocks produced for specific block ranges (next, current, previous).",
 	}, []string{"block_time"})
 
-	m.invalidClusterValidation = util.NewRequestInvalidClusterValidationLabelsTotalCounter(reg, "block-builder-scheduler", util.GRPCProtocol)
+	m.invalidClusterValidation = util.NewRequestInvalidClusterValidationLabelsTotalCounter(reg, "block-builder", util.GRPCProtocol)
 	return m
 }
 

--- a/pkg/frontend/v2/frontend_scheduler_worker.go
+++ b/pkg/frontend/v2/frontend_scheduler_worker.go
@@ -85,7 +85,7 @@ func newFrontendSchedulerWorkers(
 			// track 1ms latency too and removing any bucket bigger than 1s.
 			Buckets: []float64{.001, .005, .01, .025, .05, .1, .25, .5, 1},
 		}, []string{schedulerAddressLabel}),
-		invalidClusterValidation: util.NewRequestInvalidClusterValidationLabelsTotalCounter(reg, "query-scheduler", util.GRPCProtocol),
+		invalidClusterValidation: util.NewRequestInvalidClusterValidationLabelsTotalCounter(reg, "query-frontend", util.GRPCProtocol),
 	}
 
 	var err error

--- a/pkg/querier/worker/scheduler_processor.go
+++ b/pkg/querier/worker/scheduler_processor.go
@@ -71,7 +71,7 @@ func newSchedulerProcessor(cfg Config, handler RequestHandler, log log.Logger, r
 			Buckets: prometheus.ExponentialBuckets(0.001, 4, 6),
 		}, []string{"operation", "status_code"}),
 
-		invalidClusterValidation: util.NewRequestInvalidClusterValidationLabelsTotalCounter(reg, "query-frontend", util.GRPCProtocol),
+		invalidClusterValidation: util.NewRequestInvalidClusterValidationLabelsTotalCounter(reg, "query-scheduler-processor", util.GRPCProtocol),
 	}
 
 	frontendClientsGauge := promauto.With(reg).NewGauge(prometheus.GaugeOpts{

--- a/pkg/scheduler/scheduler.go
+++ b/pkg/scheduler/scheduler.go
@@ -160,7 +160,7 @@ func NewScheduler(cfg Config, limits Limits, log log.Logger, registerer promethe
 		},
 		[]string{"query_component"},
 	)
-	s.invalidClusterValidation = util.NewRequestInvalidClusterValidationLabelsTotalCounter(registerer, "query-frontend", util.GRPCProtocol)
+	s.invalidClusterValidation = util.NewRequestInvalidClusterValidationLabelsTotalCounter(registerer, "query-scheduler", util.GRPCProtocol)
 
 	s.requestQueue, err = queue.NewRequestQueue(
 		s.log,

